spark-plugins/src/main/java/io/cdap/plugin/spark/DataCacher.java

```diff
@@ -0,0 +1,104 @@
+import io.cdap.cdap.api.annotation.Plugin;
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.api.plugin.PluginConfig;
+import io.cdap.cdap.etl.api.FailureCollector;
+import io.cdap.cdap.etl.api.PipelineConfigurer;
+import io.cdap.cdap.etl.api.StageConfigurer;
+import io.cdap.cdap.etl.api.batch.SparkCompute;
+import io.cdap.cdap.etl.api.batch.SparkExecutionPluginContext;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.storage.StorageLevel;
+
+import java.util.Arrays;
+
+/**
+ * SparkCompute that caches a RDD
+ */
+@Plugin(type = SparkCompute.PLUGIN_TYPE)
+@Name("DataCacher")
+@Description("Spark Data Cacher caches any incoming records and outputs them unchanged.")
+public class DataCacher extends SparkCompute<StructuredRecord, StructuredRecord> {
+
+  private final DataCacherConfig config;
+
+  public DataCacher(DataCacherConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    StageConfigurer stageConfigurer = pipelineConfigurer.getStageConfigurer();
+    stageConfigurer.setOutputSchema(stageConfigurer.getInputSchema());
+
+    config.validate(stageConfigurer.getFailureCollector());
+  }
+
+  @Override
+  public JavaRDD<StructuredRecord> transform(SparkExecutionPluginContext sparkExecutionPluginContext,
+                                             JavaRDD<StructuredRecord> javaRDD) throws Exception {
+
+    StorageLevel storageLevel = StorageLevel.MEMORY_AND_DISK();
+    if (config.storageLevel.isEmpty()) {
+      StorageLevel.fromString(config.storageLevel);
+    }
+    if (storageLevel == StorageLevel.NONE()) {
+      throw new RuntimeException(String.format("Invalid storage level '%s'. Please select a valid value"));
+    }
+
+    javaRDD.persist(storageLevel);
+    return javaRDD;
+  }
+
+  /**
+   * Config class for DataCacher.
+   */
+  public static class DataCacherConfig extends PluginConfig {
+
+    private static final String STORAGE_LEVEL = "storageLevel";
+
+
+    @Name(STORAGE_LEVEL)
+    @Description("Spark storage level used to cache the data")
+    @Macro
+    private String storageLevel;
+
+    public void validate(FailureCollector collector) {
+      if (containsMacro(STORAGE_LEVEL)) {
+        return;
+      }
+
+      String[] allowedStrings = {"DISK_ONLY", "DISK_ONLY_2", "MEMORY_ONLY", "MEMORY_ONLY_2", "MEMORY_ONLY_SER",
+        "MEMORY_ONLY_SER_2", "MEMORY_AND_DISK", "MEMORY_AND_DISK_2", "MEMORY_AND_DISK_SER", "MEMORY_AND_DISK_SER_2"};
+
+      if (storageLevel.isEmpty()) {
+        collector.addFailure("Storage level is a required value. ", "Please select the desired storage level.")
+                 .withConfigProperty(STORAGE_LEVEL);
+      } else if (!Arrays.asList(allowedStrings).contains(storageLevel.toUpperCase())) {
+        collector
+          .addFailure("Invalid value for Storage Level ", "Please select one of the valid values from the dropdown.")
```

Comment:
Instead of saying dropdown, it would be better to have the list of allowed values in the suggestive action, since this API can be used in REST API (imagine when someone export and import a pipeline, and due to future version change, some of the value might no longer be valid).

Reply:
done